nixos/modules/security/rtkit.nix

```diff
@@ -32,6 +32,23 @@ with lib;
 
     systemd.packages = [ pkgs.rtkit ];
 
+    # Stop spamming the log and fix sd_notify().
+    # This bind-mounts the notify socket so the `Status` line
+    # in `systemctl status` work which makes the constant debug
+    # logging redundant.
+    # See also:
+    # - https://github.com/heftig/rtkit/issues/22
+    # - https://github.com/heftig/rtkit/issues/27
+    systemd.services.rtkit-daemon = {
+      environment.NOTIFY_SOCKET = "/fs/sd-notify";
+      serviceConfig = {
+        BindPaths = "/run/systemd/notify:/proc/fs/sd-notify";
```

Comment:
This feels like a terrible hack. Is this the right way to go considering this service is type `dbus`, so sd_notify doesn't really change the status anyway?
See https://github.com/heftig/rtkit/blob/c295fa849f52b487be6433e69e08b46251950399/rtkit-daemon.service.in#L23
According to [service docs](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Type=) the notification is only interesting in case of `Type=notify`.
On fedora for example this works without log spam and without specific socket mapping.

Reply:
sd_notify works with non-notify services as well. The reason to use `Type=notify` is startup notifications, not whether `sd_notify()` works. That is controled by `NotifyAccess=`.

The `Status:` line is `Status: "Supervising 0 threads of 0 processes of 0 users."`, which is the same as the debug message that is logged constantly.
Fixing the status line is pretty irrelevant to me, I don't care about it at all, I just implemented it so people who care about the info that is being logged constantly have a way of accessing it that isn't annoying.

Reply:
Digging into this more, adding notification won't influence the logs, since they're both sent anyway: https://github.com/heftig/rtkit/blob/c295fa849f52b487be6433e69e08b46251950399/rtkit-daemon.c#L1435-L1446

I think bumping up the log level is a good idea, but otherwise the behaviour should be either fixed upstream or patched. The whole `chroot("/proc")` could be replaced with a strict filesystem protection at service level anyway which would unbreak the status notification. (disable [do_chroot](https://github.com/heftig/rtkit/blob/c295fa849f52b487be6433e69e08b46251950399/rtkit-daemon.c#L2160)) But messing with mounting over random `/proc` directories is just... yuck.

Reply:
Agree but it was disabling chroot() vs messing with /proc and while I agree with you that this is very ugly, it's the solution that doesn't lower security.

Reply:
Or to make it more clear: I'm not happy with this solution at all but it shuts up the daemon while not supressing info that was available previously. And upstream does not seem to care at all: https://github.com/heftig/rtkit/pull/26

Reply:
We can replicate the chroot with TemporaryFileSystem and BindPaths and then disable the chroot the application creates itself.

Reply:
Something like this should work:
```
ProtectSystem=strict
PrivateDevices=true
InaccessiblePaths=/etc
TemporaryFileSystem=/var
BindPaths=/run/systemd/notify
SystemCallFilter=~@mount
```
(not tested)